mlflow/store/artifact/optimized_s3_artifact_repo.py

```diff
@@ -88,7 +104,9 @@ def _get_region_name(self):
             s3_endpoint_url=self._s3_endpoint_url,
         )
         try:
-            head_bucket_resp = temp_client.head_bucket(Bucket=self.bucket)
+            head_bucket_kwargs = {"Bucket": self.bucket}
+            self._add_bucket_owner_if_present(head_bucket_kwargs)
+            head_bucket_resp = temp_client.head_bucket(**head_bucket_kwargs)
```

Comment:
```suggestion
            head_bucket_resp = temp_client.head_bucket(Bucket=self.bucket, ExpectedBucketOwner=self._bucket_owner)
```

this doesn't work?

Reply:
because its simpler and has fewer operations. Is there a reason we need to create a dict, mutate it, and pass it to head_bucket? Why not directly pass `ExpectedBucketOwner`?

Reply:
Yes, but there's an `if` condition in the function call. 

```
head_bucket_resp = temp_client.head_bucket(Bucket=self.bucket, ExpectedBucketOwner=self._bucket_owner)
```

will break if there is no bucket owner.